pkg/cmd/roachprod/main.go

```diff
@@ -146,7 +146,9 @@ if the user would like to update the keys on the remote hosts.
 	Run: wrap(func(cmd *cobra.Command, args []string) (retErr error) {
 		zonesMap := make(map[string][]string)
 		// Only adding aws zones because only aws.ConfigSSH uses it.
-		zonesMap[aws.ProviderName] = providerOptsContainer[aws.ProviderName].(*aws.ProviderOpts).CreateZones
+		if vm.Providers[aws.ProviderName].Active() {
+			zonesMap[aws.ProviderName] = providerOptsContainer[aws.ProviderName].(*aws.ProviderOpts).CreateZones
+		}
```

Comment:
Since `providerOptsContainer` is only valid to access if the provider is Active, and since it is only valid to try to populate providerOptsContainer after vm.Providers has been initialised. I wonder if in a follow-up we might want to make providerOptsContainer private to the roachprod package and provide APIs to access it that ensure those invariants are met.

Reply:
I was thinking of making `vm.CreateProviderOptionsContainer` check `if len(vm.Providers) == 0 {return err}` (because that implies that it was not initialized) to ensure that it's never accidentally called before initializing `vm.Providers`. For the other part, my understanding of putting providerOptsContainer behind an API is to use something like `setProviderOpt(provider string, key string, value interface)` for setting values but I imagine this will get complicated due to the `value interface` part. Is there an alternative way to achieve that?